craft_providers/bases/buildd.py

```diff
@@ -1015,6 +1016,20 @@ def _update_apt_sources(
                 details=errors.details_from_called_process_error(error),
             ) from error
 
+        # update cloud.cfg to prevent the sources.list file from being reset
+        try:
```

Comment:
Is there any time we want this to succeed if `cloud.cfg` doesn't exist?

Reply:
Great idea.  `cloud.cfg` (from `cloud-init)`is fundamental to buildd images, so I decided to raise an error if it doesn't exist.